apps/api/src/app/topics/use-cases/create-topic/create-topic.use-case.ts

```diff
@@ -47,4 +82,42 @@ export class CreateTopicUseCase {
       subscribers: [],
     };
   }
+
+  private isValidTopicKey(key: string): boolean {
+    return key.length > 0 && key.match(TOPIC_KEYS_REGEX) !== null;
+  }
+
+  private async validateTopicKey({
+    key,
+    userId,
+    environment,
+    organization,
+  }: {
+    key: string;
+    environment?: EnvironmentEntity;
+    organization?: OrganizationEntity;
+    userId: string;
+  }): Promise<void> {
+    const isDryRun = await this.featureFlagService.getFlag({
+      environment,
+      organization,
+      user: { _id: userId } as UserEntity,
+      key: FeatureFlagsKeysEnum.IS_TOPIC_KEYS_VALIDATION_DRY_RUN_ENABLED,
+      defaultValue: true,
+    });
+
+    const isValidKey = this.isValidTopicKey(key);
+
+    if (!isValidKey) {
```

Comment:
To make the code cleaner you can exit fast and the check for the dryRun using a simple if/else
```suggestion
    if (this.isValidTopicKey(key)) {
      return
    }
```